src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -582,56 +604,51 @@ def build_container_from_source_with_acr_task(self, image_name, source):
         image_name_with_tag = image_name + ":{}".format(tag_now_suffix)
-        self.image = self.registry_server + "/" + image_name_with_tag
-
-        if build_from_source:
-            logger.warning("No dockerfile detected. Attempting to build a container directly from the provided source...")
-
-            try:
-                # First try to build source using buildpacks
-                # Temporary fix: using run time tag as customer image tag
-                # Waiting for buildpacks side to fix this issue: https://github.com/buildpacks/pack/issues/1750
-                logger.warning("Attempting to build image using buildpacks...")
-                buildpack_image_name_with_tag = image_name_with_tag
-                run_image_tag = get_latest_buildpack_run_tag("aspnet", "7.0")
-                if run_image_tag is not None:
-                    buildpack_image_name_with_tag = f"{image_name}:{run_image_tag}-{tag_now_suffix}"
-                self.build_container_from_source_with_buildpack(buildpack_image_name_with_tag, source)
-                self.image = self.registry_server + "/" + buildpack_image_name_with_tag
-                return
-            except ValidationError as e:
-                logger.warning(f"Unable to use buildpacks to build image from source: {e}\nFalling back to ACR Task...")
-            except CLIError as e:
-                logger.error("Failed to use buildpacks to build image from source.")
-                raise e
 
-            # If we're unable to use the buildpack, build source using an ACR Task
-            # Moving tagging img to here
-            # Skipping the buildpacks scenario for now due to issues with buildpacks
-            # Add version tag for acr image
-            logger.warning("Attempting to build image using ACR Task...")
-            self.build_container_from_source_with_acr_task(image_name_with_tag, source)
-        else:
-            # Moving tagging img to here
-            # Skipping the buildpacks scenario for now due to issues with buildpacks
-            # Add version tag for acr image
+        if _has_dockerfile(source, dockerfile):
+            # ACR Task is the only way we have for now to build a Dockerfile using Docker.
+            self.create_acr_if_needed()
+            self.image = self.registry_server + "/" + image_name_with_tag
             queue_acr_build(
                 self.cmd,
                 self.acr.resource_group.name,
                 self.acr.name,
                 image_name_with_tag,
                 source,
                 dockerfile,
-                quiet,
+                False
             )
+            return False
+
+        # Only enable Cloud Build on Stage and Canary while the changes are deployed to all regions.
+        is_cloud_build_enabled = self.env.location and any(self.env.location.lower() == region for region in ["northcentralusstage", "centraluseuap", "eastus2euap"])
+        if self.should_create_acr and is_cloud_build_enabled:
```

Comment:
what is should_create_acr is true but cloud build is not enabled?

Reply:
Yes, good catch. Once we remove the `is_cloud_build_enabled` (== next week once the deployment is complete), then we wouldn't fall in this case anymore, but as of today I agree it's possible to be in a state where we reach the next steps without a container registry. I'm adding a `create_acr_if_needed` call to make sure this case won't be a problem.

Reply:
Please find it line 640. I'll remove it next week with the PR removing the is_cloud_build_enabled check